src/lib/vocab/account/account.schema.ts

```diff
@@ -18,9 +21,24 @@ export const AccountModelSchema = {
 	properties: {
 		account_id: {type: 'number'},
 		name: {type: 'string'},
+		settings: {$ref: '/schemas/AccountSettings.json'},
 		created: {type: 'object', instanceof: 'Date', tsType: 'Date'},
 		updated: {anyOf: [{type: 'object', instanceof: 'Date', tsType: 'Date'}, {type: 'null'}]},
 	},
-	required: ['account_id', 'name', 'created', 'updated'],
+	required: ['account_id', 'name', 'settings', 'created', 'updated'],
+	additionalProperties: false,
+};
+
+export const AccountSettingsSchema = {
+	$id: '/schemas/AccountSettings.json',
+	type: 'object',
+	properties: {
+		darkmode: {type: 'boolean'},
```

Comment:
```suggestion
		theme: {anyOf: [{type: 'string', enum: ['light', 'dark']}, {type: 'object'}]},
```

I think `theme` is better for flexibility, this way we can add any app-supported strings in the future beyond `'light` | 'dark'`, and also have a custom object, something like `{extends: 'light', vars: {color: '#123'}}`, the details of which we can figure out later. Implementing that wouldn't take much effort at all. Renaming this now means we won't need a migration.

Reply:
I think for now, leaving darkmode as a place holder should be okay (especially since we're not asserting any required fields on the schema). That way we can leave the type def for the setting up to the actual PR to implement it.

I could also just blow away darkmode as a property on the schema.

Reply:
Sounds good, I forgot since we can edit the JSON directly we can fix any issues ourselves.